superx/information_extractors/branch_info_extractor.py

```diff
@@ -0,0 +1,142 @@
+                                                  'attr_path': '{http://www.sap.com/abapxml}values/STORES',
+                                                  'chain_id': 7290027600007,
+                                                  'attrs': {'store': 'STORE', 'store_id': 'STOREID',
+                                                            'store_name': 'STORENAME',
+                                                            'address': 'ADDRESS', 'city': 'CITY'}
+                                                  },
+                                    'victory': {'store_name': 'victory',
+                                                'url': 'http://matrixcatalog.co.il/',
+                                                'needs_web_scraping': True,
+                                                'need_zip_prefix': True,
+                                                'encoding': 'UTF-8',
+                                                'link_attrs_name': 'StoresFull7290696200003',
+                                                'attr_path': 'Branches',
+                                                'chain_id': 7290696200003,
+                                                'attrs': {'store': 'Branch', 'store_id': 'StoreID',
+                                                          'store_name': 'StoreName',
+                                                          'address': 'Address', 'city': 'City'}
+                                                }
+                                    }
+
+    def run_branch_extractor(self):
+        for keys in self.super_specific_info:
+            self.current_super = self.super_specific_info[keys]
+
+            try:
+                if self.current_super['needs_web_scraping']:
+                    zip_link = self.get_zip_file_link()
+                    if self.current_super['need_zip_prefix']:
+                        self.current_super['url'] = self.current_super['url'] + zip_link
+                    else:
+                        self.current_super['url'] = zip_link
+
+                xml_file = self.get_xml_file()
+            except ConnectionError as ce:
+                # logging.error(f'Unable to extract from zip file with url: {zip_link}')
+                continue
+            else:
+                self.extract_info(xml_file)
+
+    def get_zip_file_link(self):
+        """
+        This method web scrapes the urls in url_list and creates a set of the gzip file links.
+        The method then sends the set to parsing
+        If connection to the url failed, moves on to next url
+        :param url_list: list of urls to extract zip files from
+
+        """
+        try:
+            page = requests.get(self.current_super['url'])
+            web_scrapper = BeautifulSoup(page.content, 'html.parser')
+        except Exception:
+            raise ConnectionError(f'Unable to retrieve zip file link for {self.current_super["store_name"]}')
```

Comment:
This basically hides the actual error. Is that what you want to do? Another option is not to catch the actual error and allow the exception to be raised on its own.

Reply:
yes, i dont need the actual error as i know what problem causes it. sending my own error allows me more control